data_steward/cdr_cleaner/cleaning_rules/recent_concept_suppression.py

```diff
@@ -0,0 +1,238 @@
+from google.cloud import bigquery
+
+LOGGER = logging.getLogger(__name__)
+
+SUPPRESSION_RULE_CONCEPT_TABLE = 'recent_concepts'
+CONCEPT_FIRST_USE_TABLE = 'concept_first_use'
+
+RETRIEVE_CONCEPT_FIRST_USES_QUERY = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{concept_first_use_table}}` AS
+    SELECT
+        c.concept_id,
+        CASE 
+            WHEN c.valid_start_date = '{{ default_valid_start_date }}' AND min_uses.min_use_date IS NOT NULL
+                THEN min_uses.min_use_date
+            ELSE
+                c.valid_start_date
+        END valid_start_date
+    FROM (
+        SELECT
+            concept_id, MIN(min_use_date) min_use_date
+        FROM (
+            {% for table_info in table_infos  %}
+            -- Get min date of unpivoted concept id fields --
+            (
+                WITH concept_id_fields AS (
+                SELECT
+                    {{ table_info['concept_id_fields'] | join(', ') }},
+                    {{ table_info['primary_datefield'] }}
+                FROM `{{project_id}}.{{dataset_id}}.{{table_info['table_name']}}` c
+                ),
+                unpivoted_concept_if_fields AS (
+                SELECT
+                    *
+                FROM concept_id_fields
+                UNPIVOT(concept_id FOR concept_id_field IN ({{ table_info['concept_id_fields'] | join(', ') }}))
+                )
+                SELECT
+                concept_id, MIN({{ table_info['primary_datefield'] }}) min_use_date
+                FROM unpivoted_concept_if_fields
+                GROUP BY concept_id
+            )
+            {% if not loop.last %}
+            UNION ALL
+            {% endif %}
+            {% endfor %}
+        ) combined
+        GROUP BY concept_id    
+    ) min_uses
+    RIGHT JOIN `{{project_id}}.{{dataset_id}}.concept` c
+        ON c.concept_id = min_uses.concept_id
+
+""")
+
+RECENT_CONCEPT_QUERY = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{concept_suppression_lookup_table}}` AS
+SELECT c.*
+FROM `{{project_id}}.{{dataset_id}}.concept` c
+JOIN `{{project_id}}.{{sandbox_id}}.{{concept_first_use_table}}` first_use
+    ON first_use.concept_id = c.concept_id
+WHERE
+  first_use.valid_start_date >= DATE_SUB(DATE('{{cutoff_date}}'), INTERVAL 1 YEAR)
+    AND vocabulary_id <> 'PPI'
+""")
+
+
+class RecentConceptSuppression(AbstractBqLookupTableConceptSuppression):
+
+    def __init__(
+        self,
+        project_id,
+        dataset_id,
+        sandbox_dataset_id,
+        cutoff_date=None,
+        table_namer=None,
+    ):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = "Suppress COVID EHR vaccine concepts."
```

Comment:
Please update this description to match the rule usage.  We're not limiting to COVID concepts anymore.